cmd/detectExecuteScan.go

```diff
@@ -224,15 +243,61 @@ func runDetect(ctx context.Context, config detectExecuteScanOptions, utils detec
 			err = errors.Wrapf(err, exitCodeMapping(utils.GetExitCode()))
 		}
 	}
-	// create Toolrecord file
-	toolRecordFileName, toolRecordErr := createToolRecordDetect(utils, "./", config, blackduckSystem)
-	if toolRecordErr != nil {
-		// do not fail until the framework is well established
-		log.Entry().Warning("TR_DETECT: Failed to create toolrecord file "+toolRecordFileName, err)
-	}
 	return err
 }
 
+func runDetectImages(ctx context.Context, config detectExecuteScanOptions, utils detectUtils, sys *blackduckSystem, influx *detectExecuteScanInflux, blackduckSystem *blackduckSystem) error {
+	cpePath := filepath.Join(GeneralConfig.EnvRootPath, "commonPipelineEnvironment")
+	imagesRaw := piperenv.GetResourceParameter(cpePath, "container", "imageNameTags.json")
+	if imagesRaw == "" {
+		log.Entry().Debugf("No images found to be scanned")
+		return nil
+	}
+
+	images := []string{}
+	err := json.Unmarshal([]byte(imagesRaw), &images)
+	if err != nil {
+		return errors.Wrap(err, "Unable to read cpe")
+	}
+
+	registryUser := piperenv.GetResourceParameter(cpePath, "container", "repositoryUsername")
```

Comment:
How `repositoryUsername` and `repositoryPassword ` are created under CPE container path.
Looks like KanikoExecute only generates `imageNameTags.json `and `registryUrl ` but not `username/password`. We referred this - https://github.com/SAP/jenkins-library/blob/master/resources/metadata/kanikoExecute.yaml#L309

Reply:
I think, that should be given by the staging service step: https://github.wdf.sap.corp/ContinuousDelivery/piper-library/blob/51fe98e7f596950034475b3e4c7f3bd417f4a9dc/resources/metadata/sapCallStagingService.yaml#L274-L276

cc @anilkeshav27 Correct?

Reply:
Thanks @loewenstein.
Tried with the combination of `kanikoExecute `and `sapCallStagingService ` For generating repositoryUsername/Password. But we are getting failure message as Wrong action. 
Could you please suggest the right configuration for generating the credentials file under CPE container path. 
cc @anilkeshav27 @c0d1ngm0nk3y

Reply:
cc @pbusko @modulo11 @phil9909

Reply:
why do we need to read a CPE value from the file system, instead we could allow new params and have cpe references  like ?  https://github.com/SAP/jenkins-library/pull/4804/files#diff-da165ed834ab60bb2fc7bc2fed1e93646b0f36264f16e79b160b4361781207ef

with the new params, we also introduce the possibility to scan images that run outside the build, if the image lives in a private registry then users can bring in their registry url, password (maybe we would need a vault reference as well) and username . will be a good selling feature as well ?

Reply:
@t-vijayan I guess this is more a question for you and your team as step maintainers. Would you like to change this to "auto-wiring"?

@anilkeshav27 As a separate question, could we leave it as is in this PR - for the sake of it already being open for a while - and this can always be introduced later in a follow-up PR if it is required?

Reply:
@loewenstein It sounds good to extend support for external registries as well. With this enhancement, the behaviour of both Mend and DetectScan will align with each other.

Reply:
I second to use another PR for any additional features

Reply:
i would leave the final decision to @vijayanjay , who owns the piper step , since `resourceRef` is the standard way to read the CPE values

Reply:
@anilkeshav27 

> why do we need to read a CPE value from the file system, instead we could allow new params and have cpe references like?

Because we need 4 values from the `cpe`: user, password, registry AND images. Since images is an array and can't be mapped (right?), it would be pointless and very confusing to expose those parameters. If we can map all 4 to the `cpe`, it would be preferable over reading it manually of course.

Reply:
yes we should be able to map all 4 cpe  the `imageNameTags` is the tricky bit , since there can be one image or multiple build, if it ties back to the `cnbBuild` / `kaniko` and if we fill this cpe https://github.com/SAP/jenkins-library/blob/4e3fa38dd302d441e6d2cc1b5bf92c2d3e904f2f/resources/metadata/cnbBuild.yaml#L369-L370  in all cases then it must be fine.

i know in `kaniko` we fill this list even in single image is built and i am assuming its the same in `cnbBuild`